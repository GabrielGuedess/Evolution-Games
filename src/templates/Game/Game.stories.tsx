import { Story, Meta } from '@storybook/react/types-6-0';

import { Game, GameProps } from './Game';
import { gameMock } from './mock';

export default {
  title: 'Templates/Game',
  component: Game,
  parameters: {
    layout: 'fullscreen',
  },
  args: gameMock,
} as Meta<GameProps>;

export const Default: Story<GameProps> = args => <Game {...args} />;
