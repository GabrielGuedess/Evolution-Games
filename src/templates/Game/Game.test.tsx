import { renderWithTheme } from 'utils/tests/helpers';

import { screen, waitFor, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Game } from './Game';
import { gameMock } from './mock';

type Image = { src: string; alt: string; width: number; height: number };

jest.mock('next/image', () => ({
  __esModule: true,
  default: (image: Image) => (
    <img
      src={image.src}
      alt={image.alt}
      width={image.width}
      height={image.height}
    />
  ),
}));

jest.mock('templates/Base/Base', () => ({
  __esModule: true,
  Base: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Base Mock">{children}</div>;
  },
}));

jest.mock('components/organisms/GameList/GameList', () => ({
  __esModule: true,
  GameList: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="GameList Mock">{children}</div>;
  },
}));

jest.mock('components/organisms/HighlightCarousel/HighlightCarousel', () => ({
  __esModule: true,
  HighlightCarousel: function Mock({
    children,
  }: {
    children: React.ReactNode;
  }) {
    return <div data-testid="HighlightCarousel">{children}</div>;
  },
}));

jest.mock('yet-another-react-lightbox', () => ({
  __esModule: true,
  default: function Mock({
    children,
    close,
  }: {
    children: React.ReactNode;
    close: () => void;
  }) {
    return (
      <div data-testid="Lightbox Mock">
        <button type="button" data-testid="close" onClick={close}>
          Close
        </button>
        {children}
      </div>
    );
  },
}));

jest.mock('yet-another-react-lightbox/plugins/fullscreen', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('yet-another-react-lightbox/plugins/thumbnails', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('yet-another-react-lightbox/plugins/zoom', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('swiper', () => ({
  __esModule: true,
  A11y: jest.fn(),
  Scrollbar: jest.fn(),
  Navigation: jest.fn(),
  Pagination: jest.fn(),
  Keyboard: jest.fn(),
}));

jest.mock('swiper/react', () => ({
  __esModule: true,
  Swiper: function Mock({ children }: { children: React.ReactNode }) {
    return (
      <div data-testid="Swiper">
        <div data-testid="Swiper-Slide">{children}</div>
      </div>
    );
  },

  SwiperSlide: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Swiper-Slide">{children}</div>;
  },
}));

jest.mock('yet-another-react-lightbox/styles.css', jest.fn());
jest.mock('yet-another-react-lightbox/plugins/thumbnails.css', jest.fn());

describe('<Game />', () => {
  it('should render components correctly', () => {
    // Arrange
    const { container } = renderWithTheme(<Game {...gameMock} />);

    // Assert
    expect(screen.getByTestId('Base Mock')).toBeInTheDocument();
    expect(screen.getAllByTestId('Swiper')).toHaveLength(2);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the lightbox on click in desktop', async () => {
    renderWithTheme(<Game {...gameMock} />);

    // Arrange
    const imageDesktop = screen.getAllByLabelText('Star Wars image gallery')[0];

    // Act
    await userEvent.click(imageDesktop);

    // Assert
    await waitFor(() => {
      expect(screen.getByTestId('Lightbox Mock')).toBeInTheDocument();
    });
  });

  it('should render the lightbox on click in mobile', async () => {
    renderWithTheme(<Game {...gameMock} />);

    // Arrange
    const imageMobile = screen.getAllByLabelText('Star Wars image gallery')[1];

    // Act
    await userEvent.click(imageMobile);

    // Assert
    await waitFor(() => {
      expect(screen.getByTestId('Lightbox Mock')).toBeInTheDocument();
    });
  });

  it('should close lightbox', async () => {
    renderWithTheme(<Game {...gameMock} />);

    // Arrange
    const imageDesktop = screen.getAllByLabelText('Star Wars image gallery')[0];

    // Act
    fireEvent.click(imageDesktop);
    fireEvent.click(screen.getByTestId('close'));

    // Assert
    expect(screen.queryByTestId('Lightbox Mock')).toBeNull();
  });
});
