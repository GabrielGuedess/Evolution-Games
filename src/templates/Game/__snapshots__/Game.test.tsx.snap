// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Game /> should render components correctly 1'] = `
.c7 {
  max-width: 165.6rem;
  margin: 0 1.6rem;
}

.c37 {
  display: none;
}

.c34 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FFFFFF;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c34:hover {
  cursor: pointer;
  color: #855aff;
}

.c35 {
  height: 1.8rem;
}

.c0 {
  position: relative;
  height: 100vh;
}

.c0 .backgroundImage {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: -1;
  -webkit-mask-image: linear-gradient( 90deg, transparent 20%, rgba(4,4,12,0.7) 80% ), linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
  mask-image: linear-gradient( 90deg, transparent 20%, rgba(4,4,12,0.7) 80% ), linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
}

.c0 .swiper {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 2rem 1rem;
}

.c1 {
  position: absolute;
  inset: 0;
}

.c2 {
  z-index: -1;
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
  -webkit-mask-image: linear-gradient(90deg,transparent 20%,rgba(4,4,12,0.7) 80%), linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
  mask-image: linear-gradient(90deg,transparent 20%,rgba(4,4,12,0.7) 80%), linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  display: block;
  object-fit: cover;
  width: 100%;
  height: 100%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 3rem;
  z-index: 10;
  margin-top: 6rem;
  margin-bottom: 6rem;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 3rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
}

.c8 .c9 {
  display: none;
}

.c12 {
  font-weight: 500;
  line-height: 3.2rem;
  font-size: 2rem;
  color: #F4F4F5;
}

.c12 span {
  font-weight: 500;
  line-height: 3.2rem;
  text-shadow: 0 0.8rem 1rem rgba(118,74,241,0.5);
  font-size: 2rem;
  color: #855aff;
}

.c20 {
  font-style: italic;
  color: #F4F4F5;
  font-size: 1.6rem;
  line-height: 3.2rem;
}

.c13 {
  line-height: 3.2rem;
  font-size: 1.4rem;
  color: #F4F4F5;
}

.c15 {
  position: absolute;
  top: 50%;
  left: -2rem;
  cursor: pointer;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c15.swiper-button-disabled {
  display: none;
}

.c17 {
  position: absolute;
  top: 50%;
  right: -2rem;
  cursor: pointer;
  -webkit-transform: translate(50%,-50%);
  -ms-transform: translate(50%,-50%);
  transform: translate(50%,-50%);
}

.c17.swiper-button-disabled {
  display: none;
}

.c24 {
  width: 2.5rem;
  height: 2.5rem;
  cursor: pointer;
}

.c25 {
  width: 2.5rem;
  height: 2.5rem;
  cursor: pointer;
}

.c14 {
  position: relative;
  overflow: visible;
}

.c16 {
  position: relative;
  width: 100%;
  aspect-ratio: 16/9;
  border-radius: 1rem;
  cursor: pointer;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  box-shadow: 0 0.8rem 1rem rgba(133,90,255,0.25);
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c21 {
  font-style: italic;
  font-size: 1.4rem;
  line-height: 3.2rem;
  color: #F4F4F5;
}

.c21 span {
  color: #855aff;
}

.c22 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  gap: 5rem;
  padding-bottom: calc(6rem - 1.6rem);
  z-index: 10;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4rem;
}

.c26 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2.5rem;
}

.c27 {
  font-size: 2.8rem;
  font-weight: 500;
  color: #FFFFFF;
}

.c28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c29 {
  font-size: 1.6rem;
  font-weight: 500;
  line-height: 3.2rem;
  color: #F4F4F5;
}

.c30 {
  font-size: 1.6rem;
  font-weight: 500;
  line-height: 3.2rem;
  color: #717B8A;
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c32 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.6rem;
}

.c33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 1.6rem;
}

.c36 {
  font-weight: 500;
  font-size: 2.8rem;
  color: #F4F4F5;
}

.c4 {
  display: none;
  position: absolute;
  inset: 0 auto 0 0;
  overflow: hidden;
  width: 50%;
  height: 100%;
}

.c5 {
  position: relative;
  width: 100%;
  height: 100%;
}

.c6 {
  position: absolute;
  top: -10%;
  left: -10%;
  width: 100%;
  height: 120%;
  -webkit-filter: blur(1.5rem);
  filter: blur(1.5rem);
  background: linear-gradient(90deg,#04040c 53.13%,rgba(4,4,12,0.7) 100%);
}

@media (min-width:1170px) {
  .c7 {
    margin: 0 3.2rem;
  }
}

@media (min-width:1440px) {
  .c7 {
    margin: 0 4.8rem;
  }
}

@media (min-width:1656px) {
  .c7 {
    margin-left: auto;
    margin-right: auto;
  }
}

@media (max-width:1170px) {
  .c37 {
    display: block;
  }
}

@media (min-width:1170px) {
  .c0 .backgroundImage {
    -webkit-mask-image: linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
    mask-image: linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
  }
}

@media (min-width:1170px) {
  .c2 {
    -webkit-mask-image: linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
    mask-image: linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
  }
}

@media (min-width:1170px) {
  .c18 {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 1rem;
  }
}

@media (min-width:1170px) {
  .c8 {
    padding-top: 10rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    height: 100%;
    margin-left: 0;
  }

  .c8 .c9 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    position: relative;
    width: 50%;
    height: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 3rem;
    padding: 0 12% 2rem 0;
    overflow-y: auto;
    z-index: 10;
    margin-top: 0;
    margin-bottom: 0;
    padding-left: 3.2rem;
  }

  .c8 .c9::-webkit-scrollbar {
    display: none;
  }
}

@media (min-width:1440px) {
  .c8 .c9 {
    padding-left: 4.8rem;
  }
}

@media (min-width:1656px) {
  .c8 {
    margin-left: calc(((100% - 165.6rem) / 2) - 4.8rem);
  }
}

@media (min-width:1170px) {
  .c12 {
    font-size: 2.5rem;
  }

  .c12 span {
    font-size: 2.5rem;
  }
}

@media (min-width:1170px) {
  .c13 {
    font-size: 1.6rem;
  }
}

@media (min-width:1170px) {
  .c24 {
    width: 3rem;
    height: 3rem;
  }
}

@media (min-width:1170px) {
  .c25 {
    width: 3rem;
    height: 3rem;
  }
}

@media screen and (orientation:landscape) {
  .c22 {
    gap: 1rem;
    padding-bottom: calc(3rem - 1.6rem);
  }
}

@media (min-width:1170px) {
  .c22 {
    width: 50%;
    gap: 5rem;
    padding-bottom: calc(9rem - 1.6rem);
  }
}

@media screen and (orientation:landscape) {
  .c23 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    gap: 2rem;
    padding-right: 1.2rem;
  }
}

@media (min-width:1170px) {
  .c23 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 4rem;
    padding-right: 0;
  }
}

@media screen and (orientation:landscape) {
  .c26 {
    gap: 1.5rem;
  }
}

@media (min-width:1440px) {
  .c26 {
    gap: 3rem;
  }
}

@media (min-width:1170px) {
  .c27 {
    font-size: 4.5rem;
  }
}

@media (min-width:1170px) {
  .c29 {
    font-size: 2rem;
  }
}

@media (min-width:1170px) {
  .c30 {
    font-size: 2rem;
  }
}

@media (min-width:1170px) {
  .c36 {
    font-size: 3.2rem;
  }
}

@media (min-width:1170px) {
  .c4 {
    display: block;
  }
}

<div
  data-testid="Base Mock"
>
  <div
    class=""
  >
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <video
            autoplay=""
            class="c3"
            loop=""
          >
            <source
              src="https://gmedia.playstation.com/is/content/SIEPDC/global_pdc/en/games/pdps/s/sp/spiderman-miles-morales/video/spider-man-miles-morales-hero-video-01-15sec-720-web-en-06jul20.mp4"
            />
          </video>
        </div>
      </div>
      <div
        class="0 c4"
      >
        <div
          class="1 c5"
        >
          <div
            class="2 c6"
          />
        </div>
      </div>
      <div
        class="c7 c8"
      >
        <div
          class="c9 c10"
        >
          <div
            class="c11"
          >
            <strong
              class="c12"
            >
              Sobre esse 
              <span>
                Jogo
              </span>
            </strong>
            <span
              class="1 c13"
            >
              The Witcher: Wild Hunt é um RPG de mundo aberto de fantasia cheio de escolhas vitais. Em The Witcher, você joga como um caçador de monstros profissional, Geralt de Rívia, em busca da criança da profecia em um vasto mundo aberto, rico em cidades mercantis, ilhas piratas, passagens perigosas nas montanhas e cavernas esquecidas a serem exploradas.
            </span>
          </div>
          <div
            class="c11"
          >
            <strong
              class="c12"
            >
              Galeria
            </strong>
            <div
              class="6 c14"
            >
              <svg
                class="2 c15"
                fill="currentColor"
                height="25"
                id="left"
                viewBox="0 0 256 256"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <polyline
                  fill="none"
                  points="160 208 80 128 160 48"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="16"
                />
              </svg>
              <div
                data-testid="Swiper"
              >
                <div
                  data-testid="Swiper-Slide"
                >
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="The Witcher image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="The Witcher image gallery"
                        src="/img/games/god.jpg"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="Resident Evil 8 image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="Resident Evil 8 image gallery"
                        src="/img/games/r8.jpg"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="Star Wars image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="Star Wars image gallery"
                        src="/img/games/starWars.jpg"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="The Last Of Us image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="The Last Of Us image gallery"
                        src="/img/games/theLastOfUs.jpg"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="Star Wars Highlight image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="Star Wars Highlight image gallery"
                        src="/img/games/starWarsHighlight.png"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="The Witcher 2 image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="The Witcher 2 image gallery"
                        src="/img/games/god.jpg"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <svg
                class="3 c17"
                fill="currentColor"
                height="25"
                id="right"
                viewBox="0 0 256 256"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <polyline
                  fill="none"
                  points="96 48 176 128 96 208"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="16"
                />
              </svg>
            </div>
          </div>
          <div
            class="c11"
          >
            <strong
              class="c12"
            >
              Requisitos de Sistema
            </strong>
            <div
              class="c18"
            >
              <div
                class="8 c19"
              >
                <span
                  class="0 c20"
                >
                  Mínimos:
                </span>
                <p
                  class="9 c21"
                >
                  <span>
                    SO:
                  </span>
                   
                  SO: 64-bit Windows 7, 64-bit Windows 8 (8.1) or 64-bit Windows 10
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Processador:
                  </span>
                   
                  Intel CPU Core i5-2500K 3.3GHz / AMD CPU Phenom II X4 940
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Memória:
                  </span>
                   
                  6 GB de RAM
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Placa de vídeo:
                  </span>
                   
                  Nvidia GPU GeForce GTX 660 / AMD GPU Radeon HD 7870
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Armazenamento:
                  </span>
                   
                  35 GB de espaço disponível
                </p>
              </div>
              <div
                class="8 c19"
              >
                <span
                  class="0 c20"
                >
                  Recomendados:
                </span>
                <p
                  class="9 c21"
                >
                  <span>
                    SO:
                  </span>
                   
                  SO: 64-bit Windows 7, 64-bit Windows 8 (8.1) or 64-bit Windows 11
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Processador:
                  </span>
                   
                  Intel CPU Core i7 3770 3.4 GHz / AMD CPU AMD FX-8350 4 GHz
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Memória:
                  </span>
                   
                  8 GB de RAM
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Placa de vídeo:
                  </span>
                   
                  Nvidia GPU GeForce GTX 770 / AMD GPU Radeon R9 290
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Armazenamento:
                  </span>
                   
                  35 GB de espaço disponível
                </p>
              </div>
            </div>
          </div>
          <div
            class="c11"
          >
            <strong
              class="c12"
            >
              Plataformas Disponíveis
            </strong>
            <span
              class="0 c20"
            >
              Computador Pessoal
            </span>
          </div>
        </div>
        <div
          class="0 c22"
        >
          <div
            class="1 c23"
          >
            <svg
              class="4 c24"
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M128,216S28,160,28,92A52,52,0,0,1,128,72h0A52,52,0,0,1,228,92C228,160,128,216,128,216Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
            <svg
              class="5 c25"
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M208.8,72H47.2a8.1,8.1,0,0,0-8,7.1L25,207.1a8,8,0,0,0,7.9,8.9H223.1a8,8,0,0,0,7.9-8.9l-14.2-128A8.1,8.1,0,0,0,208.8,72Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M88,72a40,40,0,0,1,80,0"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
          </div>
          <div
            class="2 c26"
          >
            <h1
              class="3 c27"
            >
              The Witcher 3: Wild Hunt
            </h1>
            <div
              class="4 c28"
            >
              <span
                class="5 c29"
              >
                RPG, Action
              </span>
              <span
                class="5 c30"
                color="secondary"
              >
                CD Project
                , 
                2019
              </span>
            </div>
            <div
              class="6 c31"
            >
              <div
                class="7 c32"
              >
                <div
                  class="8 c33"
                >
                  <div
                    class="c34"
                    color="white"
                  >
                    <svg
                      aria-label="PC Icon"
                      class="c35"
                      fill="none"
                      role="figure"
                      viewBox="0 0 28 16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M.355 15.5 2.77.955h5.739c1.098 0 2 .21 2.705.632.71.416 1.208.996 1.492 1.74.289.738.355 1.59.199 2.557-.161.97-.511 1.825-1.051 2.563a5.464 5.464 0 0 1-2.088 1.72c-.857.411-1.847.617-2.97.617H3.14l.405-2.464h3.16c.597 0 1.106-.102 1.528-.306a2.581 2.581 0 0 0 1.015-.86 3.09 3.09 0 0 0 .49-1.27c.08-.488.055-.912-.078-1.272a1.559 1.559 0 0 0-.731-.838c-.35-.203-.822-.305-1.414-.305H5.44L3.43 15.5H.355Zm27.338-9.453h-3.068a2.949 2.949 0 0 0-.156-1.08 2.11 2.11 0 0 0-.511-.81 2.146 2.146 0 0 0-.817-.504 3.133 3.133 0 0 0-1.087-.177c-.776 0-1.489.194-2.137.582-.65.389-1.196.952-1.641 1.69-.445.734-.753 1.622-.923 2.664-.166 1.004-.157 1.844.028 2.521.185.677.511 1.189.98 1.534.474.341 1.063.512 1.769.512.435 0 .85-.055 1.242-.164.393-.113.751-.274 1.073-.483.327-.213.61-.47.852-.774.246-.303.438-.646.575-1.03h3.09a7.033 7.033 0 0 1-.91 1.932 7.273 7.273 0 0 1-1.512 1.662 7.3 7.3 0 0 1-2.052 1.15c-.767.285-1.608.427-2.522.427-1.32 0-2.455-.303-3.402-.91-.942-.605-1.621-1.479-2.038-2.62-.417-1.141-.495-2.517-.235-4.126.261-1.558.77-2.88 1.527-3.963.763-1.09 1.688-1.916 2.778-2.48a7.553 7.553 0 0 1 3.508-.844c.867 0 1.65.118 2.35.355.702.237 1.298.582 1.79 1.037.498.45.872 1.001 1.123 1.655.25.653.36 1.401.326 2.244Z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <strong
                class="9 c36"
              >
                R$ 60,00
              </strong>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c37"
    >
      <div
        class="c7"
      >
        <div
          class="c9 c10"
        >
          <div
            class="c11"
          >
            <strong
              class="c12"
            >
              Sobre esse 
              <span>
                Jogo
              </span>
            </strong>
            <span
              class="1 c13"
            >
              The Witcher: Wild Hunt é um RPG de mundo aberto de fantasia cheio de escolhas vitais. Em The Witcher, você joga como um caçador de monstros profissional, Geralt de Rívia, em busca da criança da profecia em um vasto mundo aberto, rico em cidades mercantis, ilhas piratas, passagens perigosas nas montanhas e cavernas esquecidas a serem exploradas.
            </span>
          </div>
          <div
            class="c11"
          >
            <strong
              class="c12"
            >
              Galeria
            </strong>
            <div
              class="6 c14"
            >
              <div
                data-testid="Swiper"
              >
                <div
                  data-testid="Swiper-Slide"
                >
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="The Witcher image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="The Witcher image gallery"
                        src="/img/games/god.jpg"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="Resident Evil 8 image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="Resident Evil 8 image gallery"
                        src="/img/games/r8.jpg"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="Star Wars image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="Star Wars image gallery"
                        src="/img/games/starWars.jpg"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="The Last Of Us image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="The Last Of Us image gallery"
                        src="/img/games/theLastOfUs.jpg"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="Star Wars Highlight image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="Star Wars Highlight image gallery"
                        src="/img/games/starWarsHighlight.png"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="Swiper-Slide"
                  >
                    <div
                      aria-label="The Witcher 2 image gallery"
                      class="7 c16"
                    >
                      <img
                        alt="The Witcher 2 image gallery"
                        src="/img/games/god.jpg"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c11"
          >
            <strong
              class="c12"
            >
              Requisitos de Sistema
            </strong>
            <div
              class="c18"
            >
              <div
                class="8 c19"
              >
                <span
                  class="0 c20"
                >
                  Mínimos:
                </span>
                <p
                  class="9 c21"
                >
                  <span>
                    SO:
                  </span>
                   
                  SO: 64-bit Windows 7, 64-bit Windows 8 (8.1) or 64-bit Windows 10
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Processador:
                  </span>
                   
                  Intel CPU Core i5-2500K 3.3GHz / AMD CPU Phenom II X4 940
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Memória:
                  </span>
                   
                  6 GB de RAM
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Placa de vídeo:
                  </span>
                   
                  Nvidia GPU GeForce GTX 660 / AMD GPU Radeon HD 7870
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Armazenamento:
                  </span>
                   
                  35 GB de espaço disponível
                </p>
              </div>
              <div
                class="8 c19"
              >
                <span
                  class="0 c20"
                >
                  Recomendados:
                </span>
                <p
                  class="9 c21"
                >
                  <span>
                    SO:
                  </span>
                   
                  SO: 64-bit Windows 7, 64-bit Windows 8 (8.1) or 64-bit Windows 11
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Processador:
                  </span>
                   
                  Intel CPU Core i7 3770 3.4 GHz / AMD CPU AMD FX-8350 4 GHz
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Memória:
                  </span>
                   
                  8 GB de RAM
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Placa de vídeo:
                  </span>
                   
                  Nvidia GPU GeForce GTX 770 / AMD GPU Radeon R9 290
                </p>
                <p
                  class="9 c21"
                >
                  <span>
                    Armazenamento:
                  </span>
                   
                  35 GB de espaço disponível
                </p>
              </div>
            </div>
          </div>
          <div
            class="c11"
          >
            <strong
              class="c12"
            >
              Consoles Disponíveis
            </strong>
            <span
              class="0 c20"
            >
              Computador Pessoal
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="HighlightCarousel"
    />
  </div>
</div>
`;
